doc/classes/Logger.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="Logger" inherits="RefCounted" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+		Custom logger to receive messages from the internal error/warning stream.
+	</brief_description>
+	<description>
+		Custom logger to receive messages from the internal error/warning stream. Loggers are registered via [method OS.add_logger].
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="_log_error" qualifiers="virtual">
+			<return type="void" />
+			<param index="0" name="function" type="String" />
+			<param index="1" name="file" type="String" />
+			<param index="2" name="line" type="int" />
+			<param index="3" name="code" type="String" />
+			<param index="4" name="rationale" type="String" />
+			<param index="5" name="editor_notify" type="bool" />
+			<param index="6" name="error_type" type="int" />
+			<param index="7" name="script_backtraces" type="ScriptBacktrace[]" />
+			<description>
+				Called when an error is logged. The error provides the [param function], [param file] and [param line] that it originated from, as well as either the [param code] that generated the error or a [param rationale].
+				The type of error provided by [param error_type] is described in the [enum ErrorType] enumeration.
+				Additionally, in debug builds, the [param script_backtraces] will be made available to this function.
+				[b]Warning:[/b] This function may be called from multiple different threads, so you may need to do your own locking.
+			</description>
+		</method>
+		<method name="_log_message" qualifiers="virtual">
+			<return type="void" />
+			<param index="0" name="message" type="String" />
+			<param index="1" name="error" type="bool" />
+			<description>
+				Called when any message is logged, including errors. If [param error] is [code]true[/code], then this message was meant to be sent to [code]stderr[/code].
```

Comment:
> Called when any message is logged, including errors.

`push_error` and `push_warning` do not call `_log_message`, and sent via `_log_error`, so this is not exactly all logged messages, only all `print*` messages.

Reply:
I don't quite know what to write here. The ambiguity between `::OS::print_error` (which turns into `::Logger::log_error`) and `::OS::printerr` (which turns into `::Logger::logv`) is quite confusing.

I changed it to just say:

> Called when a message is logged. If [param error] is [code]true[/code], then this message was meant to be sent to [code]stderr[/code].

But this will no doubt leave people wondering why `[param error]` exists when there's already a `_log_error`.